paddle/fluid/operators/optimizers/adam_op.h

```diff
@@ -372,6 +373,28 @@ class AdamOpKernel : public framework::OpKernel<T> {
             functor.adam_update(i, grad_data[row_index * row_numel + offset]);
           }
         }
+      } else if (FLAGS_inner_op_parallelism > 1 &&
+                 FLAGS_min_param_size_to_use_multithread > 0 &&
+                 param.numel() > FLAGS_min_param_size_to_use_multithread) {
```

Comment:
`numel()` is slow, cache it to a var should be better

Reply:
removed